pkg/loki/query.go

```diff
@@ -273,14 +331,106 @@ func (q *Query) processLineFilters(key string, values []string) error {
 	}
 
 	if regexStr.Len() > 0 {
-		q.lineFilters = append(q.lineFilters, regexStr.String())
+		if q.currentGroup == nil {
+			q.lineFilters = append(q.lineFilters, regexStr.String())
+		} else {
+			lf, ok := lineToLabelFilter(regexStr.String())
+			if ok {
+				q.groupedLabelFilters[*q.currentGroup] = append(q.groupedLabelFilters[*q.currentGroup], lf)
+			} else {
+				qlog.WithField("lineFilter", lf).
+					Warningf("line filter can't be parsed as json attribute. Ignoring it")
+			}
+		}
 	}
 	return nil
 }
 
+func (q *Query) processCommonLabelFilter(key string, values []string) {
+	for _, value := range values {
+		regexStr := strings.Builder{}
+		// match start any if not quoted
+		if !strings.HasPrefix(value, `"`) {
+			regexStr.WriteString(".*")
+		}
+		//inject value with regex
+		regexStr.WriteString(valueReplacer.Replace(value))
+		// match end any if not quoted
+		if !strings.HasSuffix(value, `"`) {
+			regexStr.WriteString(".*")
+		}
+		// apply filter on both Src and Dst fields
+		if q.currentGroup == nil {
+			q.labelFilters = append(q.labelFilters, regexLabelFilter("Src"+key, labelMatches, regexStr.String()))
+			q.labelFilters = append(q.labelFilters, regexLabelFilter("Dst"+key, labelMatches, regexStr.String()))
+		} else {
+			q.groupedLabelFilters[*q.currentGroup] = append(q.groupedLabelFilters[*q.currentGroup], regexLabelFilter("Src"+key, labelMatches, regexStr.String()))
+			q.groupedLabelFilters[*q.currentGroup] = append(q.groupedLabelFilters[*q.currentGroup], regexLabelFilter("Dst"+key, labelMatches, regexStr.String()))
+		}
+	}
+}
+
+func (q *Query) processFQDNFilter(key string, values []string) {
+	prefix := ""
+	if strings.HasPrefix(key, "Src") {
+		prefix = "Src"
+	} else if strings.HasPrefix(key, "Dst") {
+		prefix = "Dst"
+	}
+
+	for _, value := range values {
```

Comment:
It could be risky to handle so many different cases under the "FQDN" case, no? For instance, what if we're doing a partial matches? could "10.9" be seen as namespace + pod, whereas we're looking for an IP with partial match?

I just wonder if you're not trying to do too much here, beyond the main use case of a FQDN which would just be "namespace.pod" ?
Oh by the way, I think it should be reversed, "pod.namespace" rather than "namespace.pod", it more on page with the actual kube domain naming: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields

Reply:
Yes there are a lot of cases but I guess we can improve front end input to split them correctly. As a user I feel it's easier to type namespace first. 
What if we separate `ip:port` from `namespace.pod` cases in two filters and implement autocompletes for namespaces & pods ? 
It could show a first AC for namespaces then a second one for pods (filtered by selected namespace). 
Then the concatenation will be done by the code and added to filters to avoid mistakes.

Reply:
yes, if the user is presented specific fields for namespace and pod, it solves the FQDN order problem (but it's more work to do, especially with auto-completion).
+1 also to separate ip/port to a different filter type.